src/coreclr/jit/fgwasm.h

```diff
@@ -4,6 +4,180 @@
 #ifndef FGWASM_H
 #define FGWASM_H
 
+// Forward declarations
+//
+class Scc;
+
+//------------------------------------------------------------------------------
+// WasmSuccessorEnumerator: adapter for visiting Wasm successors in a DFS
+//
+class WasmSuccessorEnumerator
+{
+    BasicBlock* m_block;
+    union
+    {
+        // We store up to 4 successors inline in the enumerator. For ASP.NET
+        // and libraries.pmi this is enough in 99.7% of cases.
+        BasicBlock*  m_successors[4];
+        BasicBlock** m_pSuccessors;
+    };
+
+    unsigned m_numSuccs;
+    unsigned m_curSucc = UINT_MAX;
+
+public:
+    // Constructs an enumerator of all `block`'s successors.
+    WasmSuccessorEnumerator(Compiler* comp, BasicBlock* block, const bool useProfile = false);
+
+    // Gets the block whose successors are enumerated.
+    BasicBlock* Block()
+    {
+        return m_block;
+    }
+
+    // Returns the next available successor or `nullptr` if there are no more successors.
+    BasicBlock* NextSuccessor()
+    {
+        m_curSucc++;
+        if (m_curSucc >= m_numSuccs)
+        {
+            return nullptr;
+        }
+
+        if (m_numSuccs <= ArrLen(m_successors))
+        {
+            return m_successors[m_curSucc];
+        }
+
+        return m_pSuccessors[m_curSucc];
+    }
+
+    // iterator support
+    // for (BasicBlock* const succ : WasmSuccessorEnumerator(...))
+
+    class Iterator
+    {
+    private:
+        WasmSuccessorEnumerator* m_enumerator;
+        BasicBlock*              m_block;
+
+        void advance()
+        {
+            if (m_enumerator != nullptr)
+            {
+                m_block = m_enumerator->NextSuccessor();
+            }
+        }
+
+    public:
+
+        Iterator(WasmSuccessorEnumerator* enumerator)
+            : m_enumerator(enumerator)
+            , m_block(nullptr)
+        {
+            advance();
+        }
+
+        BasicBlock* operator*() const
+        {
+            assert(m_block != nullptr);
+            return m_block;
+        }
+
+        Iterator& operator++()
+        {
+            advance();
+            return *this;
+        }
+
+        bool operator!=(const Iterator& other) const
+        {
+            return m_block != other.m_block;
+        }
+    };
+
+    Iterator begin()
+    {
+        return Iterator(this);
+    }
+
+    Iterator end()
+    {
+        return Iterator(nullptr);
+    }
+};
+
+//------------------------------------------------------------------------------
+// FgWasm: Wasm-specific flow graph methods
+//
+class FgWasm
+{
+private:
+
+    Compiler*         m_comp;
+    unsigned          m_sccNum;
+    FlowGraphDfsTree* m_dfsTree;
+    BitVecTraits      m_traits;
+
+public:
+
+    FgWasm(Compiler* comp)
+        : m_comp(comp)
+        , m_sccNum(0)
+        , m_dfsTree(nullptr)
+        , m_traits(0, nullptr)
+    {
+    }
+
+    Compiler* Comp() const
+    {
+        return m_comp;
+    }
+
+    unsigned GetSccNum()
```

Comment:
Nit: This reads like an accessor name, not a mutator. `GetNewSccNum` or `GetNextSccNum` would be better.

Reply:
Thanks. Will fix this.